private_sharing/forms.py

```diff
@@ -397,3 +419,32 @@ class DeleteDataFileForm(forms.Form):
     file_basename = forms.CharField(required=False, label="File basename")
 
     all_files = forms.BooleanField(required=False, label="All files")
+
+
+class SelectDatatypesForm(forms.Form):
+    """
+    A form that generates a tree for selecting datatypes.
+    """
+
+    def clean(self):
+        cleaned_data = super().clean()
+        # Check to see if anything was selected
+        # wants to be on two lines:
+        data = dict(self.data)
+        data.pop("csrfmiddlewaretoken")
+        if data:
+            cleaned_data.update(self.data)
```

Comment:
this doesn't seem to be used?

Reply:
By this point, the csrf token has been checked and is no longer necessary, and removing it makes logic later on easier.